Vates/VatesAPI/test/MDHWNexusLoadingPresenterTest.h

```diff
@@ -12,186 +23,319 @@
 #include "MantidVatesAPI/MDHWNexusLoadingPresenter.h"
 #include "MantidVatesAPI/FilteringUpdateProgressAction.h"
 
+#include <limits>
+
 using namespace Mantid::VATES;
 using namespace testing;
 //=====================================================================================
 // Functional tests
 //=====================================================================================
-class MDHWNexusLoadingPresenterTest : public CxxTest::TestSuite
-{
+class MDHWNexusLoadingPresenterTest : public CxxTest::TestSuite {
 
 private:
-
-  // Helper method to return the full path to a real nexus file that is the correct format for this functionality.
-  static std::string getSuitableFile()
-  {
-    std::string temp = Mantid::API::FileFinder::Instance().getFullPath("SEQ_MDHW.nxs");
+  // Helper method to return the full path to a real nexus file that is the
+  // correct format for this functionality.
+  static std::string getSuitableFile() {
+    std::string temp =
+        Mantid::API::FileFinder::Instance().getFullPath("SEQ_MDHW.nxs");
     return temp;
   }
-  
-  // Helper method to return the full path to a real nexus file that is the wrong format for this functionality.
-  static std::string getUnhandledFile()
-  {
-    std::string temp = Mantid::API::FileFinder::Instance().getFullPath("CNCS_7860_event.nxs");
+
+  // Helper method to return the full path to a real nexus file that is the
+  // wrong format for this functionality.
+  static std::string getUnhandledFile() {
+    std::string temp =
+        Mantid::API::FileFinder::Instance().getFullPath("CNCS_7860_event.nxs");
     return temp;
   }
 
+  vtkDataSet *doExecute(std::string filename, bool performAsserts = true) {
+    // Setup view
+    MockMDLoadingView *view = new MockMDLoadingView;
```

Comment:
Which object owns `view`? In particular, It isn't clear to me where this object (and similar ones later in this file) is deleted.

`MDHWNexusLoadingPresenter`'s constructor passes it to its base class ( `MDHWLoadingPresenter`)'s constructor where it is stored in the member variable `m_view`. Since it isn't deleted in the destructor of  `MDHWLoadingPresenter`, I'm assuming the presenter doesn't take ownership. Should it be deleted inside the test?